price-estimator/src/orderbook.rs

```diff
@@ -1,35 +1,83 @@
-use core::orderbook::StableXOrderBookReading;
-use pricegraph::Pricegraph;
-use std::time::{Duration, SystemTime};
+use crate::solver_rounding_buffer;
+use core::{
+    models::TokenId, orderbook::StableXOrderBookReading,
+    price_estimation::price_source::PriceSource,
+};
+use pricegraph::{Pricegraph, TokenPair};
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
 use tokio::sync::RwLock;
 
+pub type DynPriceSource = Box<dyn PriceSource + Send + Sync>;
+
 /// Access and update the pricegraph orderbook.
-pub struct Orderbook<T> {
-    orderbook_reading: T,
+pub struct Orderbook {
+    orderbook_reading: Arc<dyn StableXOrderBookReading>,
+    price_source: DynPriceSource,
+    all_tokens: Vec<TokenId>,
     pricegraph: RwLock<Pricegraph>,
+    rounding_buffer_factor: f64,
 }
 
-impl<T> Orderbook<T> {
-    pub fn new(orderbook_reading: T) -> Self {
+impl Orderbook {
+    pub fn new(
+        orderbook_reading: Arc<dyn StableXOrderBookReading>,
+        price_source: DynPriceSource,
+        all_tokens: Vec<TokenId>,
+        rounding_buffer_factor: f64,
+    ) -> Self {
+        assert!(rounding_buffer_factor.is_finite());
+        assert!(rounding_buffer_factor >= 0.0);
         Self {
             orderbook_reading,
+            price_source,
+            all_tokens,
             pricegraph: RwLock::new(Pricegraph::new(std::iter::empty())),
+            rounding_buffer_factor,
         }
     }
 
-    pub async fn get_pricegraph(&self) -> Pricegraph {
+    pub async fn pricegraph(&self) -> Pricegraph {
```

Comment:
Can we make it clear from the name that this pricegraph has rounding buffers applied to it?